devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,209 @@
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+	}
+}
+
+def on() {
+	resetActiveSound()
+	state.lastTriggeredSound = 1
+	response(zwave.basicV1.basicSet(value: 0xFF).format())
+}
+
+def off() {
+	response(zwave.basicV1.basicSet(value: 0x00).format())
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	zwave.basicV1.basicGet()
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : componentLabel,
+					isComponent   : true,
+					componentName : "Aeotec Doorbell 6 $endpoint",
+					componentLabel: "Aeotec Doorbell 6 $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def sendCommand(deviceDni, commands) {
+	def result = commands.collect {
+		if (it instanceof String) {
+			it
+		} else {
+			encap(it, channelNumber(deviceDni))
+		}
+	}
+	sendHubCommand(result, 100)
+}
+
+def channelNumber(String dni) {
+	dni?.split(":")[-1] as Integer
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationStatus == 0xFF) {
+		switch (cmd.event) {
+			case 0x09: //TAMPER
+				sendEvent([name: "tamper", value: "detected"])
+				break
+			case 0x01: //ON
+					if(state.lastTriggeredSound == 1 ) {
+						sendEvent([name: "alarm", value: "both"])
+					} else {
+						String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+						def child = childDevices.find { it.deviceNetworkId == childDni }
+						child?.sendEvent([name: "alarm", value: "both"])
+					}
+				break
+			case 0x00: //OFF
+					resetActiveSound()
+					sendEvent([name: "tamper", value: "clear"])
+				break
+		}
+	}
+}
+
+def setOnChild(deviceDni) {
+	resetActiveSound()
+	sendCommand(deviceDni, zwave.basicV1.basicSet(value: 0xFF))
+	state.lastTriggeredSound = deviceDni?.split(":")[-1] as Integer
```

Comment:
If the child DNI is always "${parentDNI}:${soundNumber}" why not drop the split and just do `deviceDni[-1] as Integer`?

I'd also say with the `deviceDni?.` part it is a little late to be protecting against a null deviceDni value. If you want to protect for a null deviceDni, I would protect the whole function.